extensions/spring-di/deployment/src/test/java/io/quarkus/spring/di/deployment/ListOfBeansTest.java

```diff
@@ -0,0 +1,94 @@
+package io.quarkus.spring.di.deployment;
+
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Repository;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+public class ListOfBeansTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClasses(Foo.class, ServiceAlpha.class, ServiceBravo.class, Service.class,
+                            Converter.class, ConverterAlpha.class, ConverterBravo.class));
+
+    @Inject
+    Foo foo;
+
+    @Test
+    public void testInjection() {
+        assertThat(foo.services).hasSize(2).extractingResultOf("ping").containsExactlyInAnyOrder("alpha", "bravo");
+        assertThat(foo.converters).hasSize(2).extractingResultOf("pong").containsExactlyInAnyOrder("alpha", "bravo");
+    }
+
+    @org.springframework.stereotype.Service
+    public static class Foo {
+
+        @Autowired
+        List<Service> services;
+
+        final List<Converter> converters;
+
+        @Autowired
```

Comment:
This should also work if `@Autowired` is missing, but it currently doesn't. 
This is a minor bug that we can fix if someone comes across it